spec/rubocop/cop/style/redundant_return_spec.rb

```diff
@@ -282,6 +282,92 @@ def func
     end
   end
 
+  context 'when return is inside begin-end body' do
+    let(:src) do
+      <<-RUBY.strip_indent
+        def func
+          begin
+            return 1
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+        def func
+          begin
+            return 1
+            ^^^^^^ Redundant `return` detected.
+          end
+        end
+      RUBY
+    end
+
+    it 'auto-corrects' do
+      corrected = autocorrect_source(src)
+      expect(corrected).to eq <<-RUBY.strip_indent
+        def func
+          begin
+            1
+          end
+        end
+      RUBY
+    end
+  end
+
+  context 'when rescue and return blocks present' do
+    let(:src) do
+      <<-RUBY.strip_indent
+        def func
+          1
+          2
+          return 3
+        rescue SomeException
+          return 4
+        rescue AnotherException
+          return 5
+        ensure
+          return 6
+        end
+      RUBY
+    end
+
+    it 'does register an offense when inside function block' do
+      expect_offense(<<-RUBY.strip_indent)
+        def func
+          1
+          2
+          return 3
+          ^^^^^^ Redundant `return` detected.
+        rescue SomeException
+          return 4
```

Comment:
I guess the `return` are also redundant.
Note: the `return` that is in `ensure` is not redundant.

Reply:
Oops. Fixed.

Reply:
Looks good 👍